config_utilities/include/config_utilities/config.h

```diff
@@ -255,4 +257,42 @@ inline void check(Ts... args) {
   internal::Visitor::visitCheck(CheckType(std::forward<Ts>(args)...));
 }
 
+/**
+ * @brief Update a field in a config.
+ * @note This function will update the field and check the validity of the config afterwards. If the config is invalid,
+ * the field will be reset to its original value.
+ * @param node The node containing the field to update.
+ */
+template <typename ConfigT>
+bool updateField(ConfigT& config, YAML::Node node, const std::string& field_name, const std::string& name_space = "") {
+  const auto old_value = internal::Visitor::getValues(config, true, name_space, field_name);
+  internal::Visitor::setValues(config, internal::lookupNamespace(node, name_space));
+  bool success = isValid(config);
+
+  if (!success) {
+    internal::Visitor::setValues(config, internal::lookupNamespace(old_value.data, name_space));
+  }
+
+  return success;
+}
+
+/**
+ * @brief Update a field in a config.
+ * @note This function will update the field and check the validity of the config afterwards. If the config is invalid,
+ * the field will be reset to its original value.
+ * @param field_name The name of the field to update.
+ * @param value The new value of the field.
+ */
+template <typename T, typename ConfigT>
```

Comment:
This is a bit tricky since our design choices around enums are that the config can decide how it wants to parse it. The easiest way to really support the desired behavior is probably to provide the value directly as string (since Enum<T>::toString() is not necessarily defined for all enums) and let config_utilites handle the conversion.